### PR TITLE
feat: Add github actions

### PR DESCRIPTION
**Pull request by marcpfuller**
_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
_Provide a 1-2 line brief overview of the changes submitted through the Pull Request..._

### REFERENCES
Reference URL for issue tracking (Github): **\<URL to be filled>**

- [ ] Added label to the Pull Request for easier discoverability and search

### CODE MAINTAINABILITY
- [ ] **_Commit Message meets guidelines as indicated in the URL\*_**
	- https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [ ] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approve the Pull Request for merge are required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
	- [ ] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Adopted domain specific coding standards ([SBL Coding Standards](https://github.com/tianocore-docs/Docs/raw/master/Specifications/CCS_2_1_Draft.pdf))
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Tracing output are minimized and logic
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

# _Code must act as a teacher for future developers_

### CODE REVIEW

File: .github/workflows/reviewdog.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            HADOLINT_VERSION: v1.16.3
+          run: |
+            wget -q https://github.com/hadolint/hadolint/releases/download/$HADOLINT_VERSION/hadolint-Linux-x86_64 -O /tmp/hadolint \
+            && chmod +x /tmp/hadolint
+        - name: Run Reviewdog
+          env:
+            REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            REVIEWDOG_FAIL_ON_ERROR: 'true'
+          run: |
+            echo "$secrets.GITHUB_TOKEN"
```
Comment (vli11) [inline .github/workflows/reviewdog.yaml:36]: I think you can remove this echo token

**Review by vli11 — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "1.5.0" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "1.5.0" ]
+  schedule:
+    - cron: '19 23 * * 0'
```
Comment (jim-wang-intel) [inline .github/workflows/codeql.yaml:21]: remove this?   do we need this schedule to run?  If so, why running on At 23:19 on Sunday?

File: .github/workflows/codeql.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
```
Comment (jim-wang-intel) [inline .github/workflows/codeql.yaml:36]: add `go` ?

File: .github/workflows/hadolintscan.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+on: [pull_request]
+env:
+  HADOLINT_VERSION: v1.16.3
+jobs:
+  bandit:
+    name: runner / hadolint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: install hadolint
```
Comment (jim-wang-intel) [inline .github/workflows/hadolintscan.yaml:11]:
```suggestion
      - name: Install hadolint
```

File: .github/workflows/pythonCodeStyleScan.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+name: PyCodestyle
+on: [pull_request]
+jobs:
+  bandit:
+    name: runner / PyCodestyle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: install pycodestyle
```
Comment (jim-wang-intel) [inline .github/workflows/pythonCodeStyleScan.yaml:9]:
```suggestion
      - name: Install pycodestyle
```

File: .github/workflows/reviewdog.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  reviewdog:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+        - name: git checkout
+          uses: actions/checkout@v3
+        
+        - name: install reviewdog
```
Comment (jim-wang-intel) [inline .github/workflows/reviewdog.yaml:16]:
```suggestion
        - name: Install reviewdog
```

File: .github/workflows/reviewdog.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+          run: |
+            curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s
+        - name: install bandit
```
Comment (jim-wang-intel) [inline .github/workflows/reviewdog.yaml:19]:
```suggestion
        - name: Install bandit
```

File: .github/workflows/reviewdog.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+          run: |
+            pip install bandit && bandit --version
+        - name: install pep8
+          run: |
+              sudo apt install pep8
+        - name: install hadolint
```
Comment (jim-wang-intel) [inline .github/workflows/reviewdog.yaml:25]:
```suggestion
        - name: Install hadolint
```

File: .github/workflows/shellcheck.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+name: Shellcheck
+on: [pull_request]
+jobs:
+  shellcheck:
+    name: runner / shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: install shellcheck
```
Comment (jim-wang-intel) [inline .github/workflows/shellcheck.yaml:9]:
```suggestion
      - name: Install shellcheck
```

**Comment by brian-intel** [comment]
Looks good after you address Valina's comment about the token

File: .reviewdog.yml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  bandit:
+    cmd:  'bandit -r ./ -c .bandit.yaml -f custom --msg-template "{relpath}:{line}:{col}:{test_id}:{severity}:{msg}"' 
+    errorformat: # (optional if you use `format`)
+      - "%f:%l:%c:%m" 
+    name: bandit # (optional. you can overwrite <tool-name> defined by runner key)
+    level: info # (optional. same as -level flag. [info,warning,error])
+  # hadolint:
+  #   cmd: '/tmp/hadolint Dockerfile -f json | jq -f ".rdjson.jq" -c'
+  #   format: rdjson
+  #   level: error # (optional. same as -level flag. [info,warning,error])
```
Comment (jim-wang-intel) [inline .reviewdog.yml:11]: still need this? if not, remove it

**Review by jim-wang-intel — COMMENTED** [review]
some suggestion making naming more consistence for steps

File: .reviewdog.yml (hunk 10 above)
Comment (marcpfuller) [reply .reviewdog.yml:11]: still need it. I am trying a different way to get reviewdog working. if it works then I can uncomment it and have it ran using reviewdog

**Review by vli11 — APPROVED** [review]
LGTM

File: .reviewdog.yml (hunk 10 above)
Comment (jim-wang-intel) [reply .reviewdog.yml:11]: got it

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by brian-intel — APPROVED** [review]
LGTM

**Pull request merged**